src/test/java/org/elasticsearch/index/query/SimpleIndexQueryParserTests.java

```diff
@@ -155,9 +158,54 @@
 
     private IndexQueryParserService queryParser;
 
+    private static class DummyQuery extends Query {
+
+        public boolean isFilter;
+        
+        @Override
+        public String toString(String field) {
+            return getClass().getSimpleName();
+        }
+        
+    }
+
+    public static class DummyQueryParser extends AbstractIndexComponent implements QueryParser {
```

Comment:
Having my reafactoring goggles on I was hoping it was possible to avoid introducing new Parsers/Builders, even if they are dummy ones. I see why this makes makes sense here though. Just wondering if if would be possible to wrap a simple existing Parser (like e.g. MatchAll) and have it produce the DummyQuery instead of having a completely new Builder/Parser pair we need to take care of in the refactoring later. Not very important, though.